source/macos/AdaptiveCards/AdaptiveCards/AdaptiveCard.swift

```diff
@@ -13,6 +13,31 @@ public protocol AdaptiveCardResourceResolver: AnyObject {
     func adaptiveCard(_ adaptiveCard: ImageResourceHandlerView, dimensionsForImageWith url: String) -> NSSize?
     func adaptiveCard(_ adaptiveCard: ImageResourceHandlerView, requestImageFor url: String)
     func adaptiveCard(_ adaptiveCard: NSView, attributedStringFor htmlString: String) -> NSAttributedString?
+    func adaptiveCard(_ adaptiveCard: NSView, typeAheadQueryFor query: String) async throws -> QueryResponse
+}
+
+public struct QueryResponse: Codable {
+    var status: Int
+    var queryBody: QueryBody
+    
+    enum CodingKeys: String, CodingKey {
+        case status
+        case queryBody = "body"
+    }
+    
+    public struct QueryBody: Codable {
+        var type: String
+        var value: QueryValues
+        
+        public struct QueryValues: Codable {
+            var results: [QueryResults]
+            
+            public struct QueryResults: Codable {
+                var value: String
+                var title: String
+            }
+        }
+    }
```

Comment:
Regarding the error handling case where we receive null values, please take this into consideration and implement optional variables. For more control over decoding, it is better to use a custom initializer (This will help you in minimising unexpected crashes such as typecast).

Reply:
Can we discuss this further?